tests/test_sshfsmount.cpp

```diff
@@ -188,167 +202,186 @@ struct SshfsMount : public mp::test::SftpServerTest
 
     const std::unordered_map<std::string, std::string> default_cmds{
         {"snap run multipass-sshfs.env", "LD_LIBRARY_PATH=/foo/bar\nSNAP=/baz\n"},
-        {"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -V", "FUSE library version: 3.0.0"},
-        {"id -u", "1000"},
-        {"id -g", "1000"},
-        {"pwd", "/home/ubuntu"},
+        {"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -V", "FUSE library version: 3.0.0\n"},
+        {"pwd", "/home/ubuntu\n"},
+        {"sudo /bin/bash -c 'P=\"/home/ubuntu/target\"; while [ ! -d \"$P/\" ]; do P=${P%/*}; done; echo $P/'",
+         "/home/ubuntu/\n"},
+        {"id -u", "1000\n"},
+        {"id -g", "1000\n"},
         {"sudo env LD_LIBRARY_PATH=/foo/bar /baz/bin/sshfs -o slave -o transform_symlinks -o allow_other :\"source\" "
          "\"target\"",
-         "don't care"}};
+         "don't care\n"}};
 };
-} // namespace
 
-TEST_F(SshfsMount, throws_when_sshfs_does_not_exist)
+// Mocks an incorrect return from a given command.
+struct SshfsMountFail : public SshfsMount, public testing::WithParamInterface<std::string>
 {
-    bool invoked{false};
-    auto request_exec = make_exec_that_fails_for({"sudo multipass-sshfs.env", "which sshfs"}, invoked);
-    REPLACE(ssh_channel_request_exec, request_exec);
+};
 
-    EXPECT_THROW(make_sshfsmount(), mp::SSHFSMissingError);
-    EXPECT_TRUE(invoked);
-}
+// Mocks correct execution of a given vector of commands/answers. The first string specifies the parameter with
+// which make_sshfsmount mock is called.
+struct SshfsMountExecute : public SshfsMount, public testing::WithParamInterface<std::pair<std::string, CommandVector>>
+{
+};
 
-TEST_F(SshfsMount, throws_when_unable_to_make_target_dir)
+struct SshfsMountExecuteAndFail
```

Comment:
```suggestion
struct SshfsMountExecuteAndNoFail
```
We actually want these to _not_ fail, right? Or not throw?

Reply:
Oops, right.